tests/test_manager.py

```diff
@@ -507,6 +522,177 @@ def test_add_task_assigns_due_if_exist(self):
         generated_task = self.session.query(Task).one()
         assert generated_task.due == due
 
+    def test_modify_task_modifies_project(self):
+        old_project = ProjectFactory.create()
+        new_project = ProjectFactory.create()
+        task = self.factory.create(state='open')
+        task.project = old_project
+
+        self.manager.modify(
+            fulid().fulid_to_sulid(task.id, [task.id]),
+            project_id=new_project.id
+        )
+
+        modified_task = self.session.query(Task).get(task.id)
+
+        assert modified_task.project is new_project
+
+    def test_modify_task_generates_project_if_doesnt_exist(self):
+        task = self.factory.create(state='open')
+
+        self.manager.modify(
+            fulid().fulid_to_sulid(task.id, [task.id]),
+            project_id='non_existent')
+
+        modified_task = self.session.query(Task).get(task.id)
+        project = self.session.query(Project).get('non_existent')
+
+        assert modified_task.project is project
+        assert isinstance(project, Project)
+
+    def test_modify_task_adds_tags(self):
+        tag_1 = TagFactory.create()
+        tag_2 = TagFactory.create()
+        task = self.factory.create(state='open')
```

Comment:
We are assuming that we want to only modify the `open` tasks, maybe we should create a new issue with the modification of other kind of tasks. (or if you see it trivial implement it).

Reply:
The implementation supports modifying tasks with any state if the `id` is a *fulid*, if it's a *sulid*, the status is assumed to be *open*. ¿Should the tests cover modifying tasks with an arbitrary state?

Reply:
Added test for `modify()` on task with arbitrary state in 2e857ded26c24c95a69b6099ae55a3a1071e81a5.